### DISCUSSION TITLE
Readings don't persisit

### DISCUSSION THREAD

**Discussion by kevinkey619** [bellrichm/WeeWX-MQTTSubscribe#157] [General]
Hi.

For some reasons my MQTT readings don't always persist long enough to stay in the packets with my other (non-MQTT) readings. Is there a way to fix this problem?

**Comment by bellrichm** [comment]
I'm guessing that you are running MQTTSubscribe as a service....
Is the problem that the MQTT data is in a the loop packets and not the archive record that is created from the loop packets? Or is the problem that the MQTT data arrives less frequently than the archive interval? What is your archive interval? How is the loop data accumulated? Is it the sum of loop data, average, last, first, etc?

**Reply by kevinkey619** [reply]
Yeah, I'm running it as service. I'm using a driver to get the other data. Can I use two drivers with weewx? So, this is expected behavior? Any workarounds?

**Reply by kevinkey619** [reply]
Value in question is pressure - so it's not cumulative. Do I need to configure the MQTT provider to send the readings more frequently? Currently it's every 10 seconds. Perhaps every second?

**Reply by kevinkey619** [reply]
my weewx.conf

[MQTTSubscribeService]
    # This section is for the MQTTSubscribe service.

    # Turn the service on and off.
    # Default is: true
    # Only used by the service.
    enable = true

    # The MQTT server.
    # Default is localhost.
    host = localhost

    # The port to connect to.
    # Default is 1883.
    port = 1883

    # Maximum period in seconds allowed between communications with the broker.
    # Default is 60.
    keepalive = 10

    # username for broker authentication.
    # Default is None.
    username = None

    # password for broker authentication.
    # Default is None.
    password = None

    # The binding, loop or archive.
    # Default is: loop

    binding = loop

    archive_interval = 5
    ignore_end_time = true

    # The message handler to use
    [[message_callback]]
        # The format of the MQTT payload.
        # Currently support: individual, json, keyword
        # Must be specified.
        type = individual

    # The topics to subscribe to.
    [[topics]]
        # Units for MQTT payloads without unit value.
        # Valid values: US, METRIC, METRICWX
        # Default is: US
        unit_system = METRIC

        [[[indoor/BME280_pressure]]]
            # The WeeWX name.
            # Default is the name from MQTT.
            name = pressure
            #ignore = false
            #qos = 2
            expires_after = None

**Reply by kevinkey619** [reply]
It seems to be ignoring the "expires_after". Or, do I have it in the wrong place?

**Comment by bellrichm** [comment]
We will have to look at the log. Set `debug = 1` and restart WeeWX. Since you are having trouble with the cache, run it until an archive record is missing the value after it has appeared at least once. Before attaching the log, double check it for any sensitive information.
And yes, the log will be very large.
